mmv1/products/vertexai/IndexEndpoint.yaml

```diff
@@ -97,3 +101,16 @@ properties:
       [Format](https://cloud.google.com/compute/docs/reference/rest/v1/networks/insert): `projects/{project}/global/networks/{network}`.
       Where `{project}` is a project number, as in `12345`, and `{network}` is network name.
     immutable: true
+  - !ruby/object:Api::Type::NestedObject
+    name: privateServiceConnectConfig
+    immutable: true
+    description: |-
+      Optional. Configuration for private service connect. `network` and `privateServiceConnectConfig` are mutually exclusive.
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: enablePrivateServiceConnect
+        description: If set to true, the IndexEndpoint is created without private service access.
+      - !ruby/object:Api::Type::Array
+        name: projectAllowlist
+        description: A list of Projects from which the forwarding rule will target the service attachment.
+        item_type: Api::Type::String
```

Comment:
Are these two subfields also immutable? If so, I think we need to add immutable as well. If not, it's be nice that we add an update test on it.

Plus, can we make at least one of the subfields required. We 'd try to avoid users from sending empty object.

Reply:
The subfields were immutable although I tested a PATCH request. Based on your advice, I added `immutable:` and `required:` in [debd2d1](https://github.com/GoogleCloudPlatform/magic-modules/pull/8851/commits/debd2d1964f87a06dcb34afa77fba318d8137ff3).